_posts/2023-05-31-adding-support-for-outline-properties.md

```diff
@@ -0,0 +1,69 @@
+---
+layout:     post
+tags:       blog
+title:      Adding support for ‘outline’ properties
+date:       2023-05-31
+summary:    Guide to the ‘outline’ implementation in Layout 2020, plus the tests and spec issues we added along the way.
+categories:
+---
+
+As mentioned in [our last blog post](https://servo.org/blog/2023/04/13/layout-2013-vs-2020/), we're currently working on selecting a layout engine for Servo between the original Layout 2013 and the newer Layout 2020.
+
+Our plan has been to start by implementing some small features in Layout 2020, to help us decide whether to switch to the new layout engine, and in turn tackle more complex features like floats. One of these features was ‘outline’, which is now supported in the new engine.
+
+A few days ago, we landed support for ‘outline’ and ‘outline-offset’. These properties are now fully supported in Servo, with two minor caveats:
+* [Snap border and outline widths at computed-value time](https://github.com/servo/servo/pull/29718) — this is blocked on a Stylo upgrade to avoid diverging from Firefox
+* The ‘outline-style’ value ‘auto’ currently works like ‘solid’ — this is allowed by the spec, but we may be able to do something better here, like rounding the corners of the outline or matching the platform style
+
+The impact of this feature is most noticeable in the focus styles for links and input fields. For example, the User Agent stylesheet already applies ‘outline: thin dotted’ to ‘a:focus’, so clicking the first link in
+
+```
```

Comment:
```suggestion
```html
```

Reply:
There seems to be some styling issue so we cannot do that. Maybe that can be fixed later at some point.

Reply:
ah, I see. thanks for filing the issue!